### DISCUSSION TITLE
Information about QSL status

### DISCUSSION THREAD

**Discussion by R1BLH** [magicbug/Cloudlog#1751] [General]
There are three fields with descriprion of the QSL status on the QSO info plane (see screenshots).
But these fields has a strange different behaviour.
1. **QSL info**
Sometimes it dispays information about "QSL has been sent via..."  (without date) -- QSL has been sent and confirmation has been not received.
Sometimes it displays nothing except title -- QSL has been received.
It would be greate to have information about method and date of the QSL (on send and reveive). Sometimes i got electronic replies on paper QSL.

2. **LoTW**
All looks OK. 
When confirmation has been received titile and message about confirmation and confirmation date are displayed.

3. **eQSL**
All looks OK like LoTW field excwpt one thing:
If you would mark eQSL received manually (this is my case, because it is really annoing to create new station location every weekend, when i move to countryhouse -- eqsl.cc requires it, but in the cloudlog i have only two locations) confirmation date always is 1970-01-01, not the date when eQSL has been marked as received,

![2022-11-08_14-55-22](https://user-images.githubusercontent.com/16667173/200559493-ca54f957-3678-4249-91e0-b17d9b5966a9.png)

![2022-11-08_15-08-10](https://user-images.githubusercontent.com/16667173/200560404-e388e4d4-9f1f-439b-b485-1f124202e4a3.png)

**Comment by phl0** [comment]
I proposed a fix for the issue with missing QSL info via https://github.com/magicbug/Cloudlog/pull/1752 (basically a mismatch of variable names in the RCVD section).
The eQSL date issue also affects the LotW part. The date is simply not set when editing the QSO manually. I am getting my head around this and may come up with a solution.

**Comment by phl0** [comment]
And I made a patch to address the issue with 1.1.1970 (i.e. empty) dates on eQSL and LotW QSL information. Now this field is updated if status changes on editing a QSO. See https://github.com/magicbug/Cloudlog/pull/1756.

**Reply by R1BLH** [reply]
@phl0 TNX!
How to test this patch? 
//Sorry for dummy question.

**Reply by phl0** [reply]
Both have been merged to dev branch. So you will have to check that branch out in order to test.

**Reply by R1BLH** [reply]
> Both have been merged to dev branch. So you will have to check that branch out in order to test.
eQSL date information possible bug detected. Now i editing old records (adding exact OTH locator data). After this eQSL date had been changed to current date, without any changes in the eQSL status (yellow arrow on the screenshot).

I already updated from git (it's sad, russian translation gone, i tried to make another PR).

There is no information about QSL (which had been sent and received by electronic method in this case, via hamlog,online). 

![2022-11-10_15-21-28](https://user-images.githubusercontent.com/16667173/201091460-78d993a5-5af1-48cf-ae08-12d14024380d.png)
![2022-11-10_15-19-50](https://user-images.githubusercontent.com/16667173/201091463-2565136d-17c5-4a5a-ae9c-059f0e45c7cf.png)

**Reply by phl0** [reply]
I will check. IIRC it only applies to QSL states Y and N, i.e. no E or other methods. I'll check if we can correct that. Can you show the QSL section in "Edit QSO" pls? Is it sent to "Sent" via "Electronic"?

**Reply by R1BLH** [reply]
> I will check. IIRC it only applies to QSL states Y and N, i.e. no E or other methods. I'll check if we can correct that. Can you show the QSL section in "Edit QSO" pls? Is it sent to "Sent" via "Electronic"?

Easy.
I did not make any changes to the QSL states (they were all already set to Yes).


![2022-11-10_15-39-10](https://user-images.githubusercontent.com/16667173/201093900-635b5581-41d4-42ad-8127-4740720095c0.png)
![2022-11-10_15-38-54](https://user-images.githubusercontent.com/16667173/201093902-f154dae4-d95a-410b-a3db-b485e05a903d.png)
![2022-11-10_15-38-39](https://user-images.githubusercontent.com/16667173/201093903-e288431b-0dfa-4fd6-8c8a-9c7b163718cc.png)

**Reply by phl0** [reply]
Tnx I will check.

**Reply by phl0** [reply]
Added details about electronic and manager with https://github.com/magicbug/Cloudlog/pull/1763.

![Screenshot from 2022-11-10 14-43-48](https://user-images.githubusercontent.com/7112907/201108315-3bb59522-9d13-4ef0-97da-2aea2b548c5c.png)

**Reply by R1BLH** [reply]
> Added details about electronic and manager with #1763.
> 
Thank you!
Russian translation updated! :)
